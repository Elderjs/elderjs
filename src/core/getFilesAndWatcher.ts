import { EventEmitter } from 'events';
import chokidar from 'chokidar';
import path, { sep } from 'path';
import fg from 'fast-glob';

import { InternalFiles, SettingsOptions } from '../utils/types.js';
import { Stats } from 'fs';
import getUniqueId from '../utils/getUniqueId.js';
import windowsPathFix from '../utils/windowsPathFix.js';

export function hashUrl(url: string) {
  return `${url}?hash=${getUniqueId()}`;
}

export function unhashUrl(url: string) {
  return url.split('?')[0];
}

export function makeCssRelative({ file, distElder }: { file: string; distElder: string }) {
  return file ? `/${path.relative(distElder, file)}` : '';
}

export function updateFilesFn({
  files,
  settings,
  paths,
}: {
  files: Partial<InternalFiles>;
  settings: Pick<SettingsOptions, 'srcDir' | 'distDir' | 'ssrComponents' | 'clientComponents'>;
  paths: string[];
}) {
  const all = fg.sync(paths).map(windowsPathFix);

  const shortcodeFile = all.find((f) => f === path.join(settings.srcDir, './shortcodes.js'));
  const hooksFile = all.find((f) => f === path.join(settings.srcDir, './hooks.js'));

  files.all = all;
  files.publicCssFile = makeCssRelative({
    file: all.find((p) => p.includes(`assets${sep}svelte`) && p.endsWith('.css')),
    distElder: settings.distDir,
  });
  files.server = all.filter((f) => f.includes(settings.ssrComponents)).map(hashUrl);
  files.routes = all
    .filter((f) => f.includes(path.join(settings.srcDir, './routes/')) && f.toLowerCase().endsWith('route.js'))
    .map(hashUrl);
  files.hooks = hooksFile ? hashUrl(hooksFile) : '';
  files.shortcodes = shortcodeFile ? hashUrl(shortcodeFile) : '';
  // already hashed
  files.client = all.filter((f) => f.includes(settings.clientComponents));

  return files as InternalFiles;
}

type TGetFilesAndWatcher = {
  clientComponents: string;
  ssrComponents: string;
  production: boolean;
  distElder: string;
  configFiles: string[];
} & Pick<SettingsOptions, 'server' | 'build' | 'srcDir' | 'rootDir' | 'distDir'>;

export default function getFilesAndWatcher(settings: TGetFilesAndWatcher): {
  files: InternalFiles;
  watcher: EventEmitter;
} {
  const watcher = new EventEmitter();
  const paths = [
    `${settings.srcDir}/**/*`,
    `${settings.ssrComponents}/**/*.js`,
    `${settings.clientComponents}/**/*.js`,
    `${settings.distDir}/**/*.css`,
    ...settings.configFiles,
  ];

  const files = {} as InternalFiles;
  files.updateFiles = () => updateFilesFn({ settings, paths, files });
  files.updateFiles();

  if (!settings.production && settings.server) {
    // todo: add in plugin folders for Elder.js
    const chok = chokidar.watch(paths, { alwaysStat: true, usePolling: true });

    // these are unhashed.
    const chokFiles: Map<string, Stats> = new Map();

    // eslint-disable-next-line no-inner-declarations
    function handleChange(file: string) {
      const fixedFile = windowsPathFix(file);
      const f = path.relative(settings.srcDir, file);
      if (f.startsWith('routes') && !f.endsWith('svelte')) {
        if (f.endsWith('route.js')) {
          if (!files.routes.includes(file)) files.routes.push(file);
          watcher.emit('route', hashUrl(file));
        } else {
          // find nearest route.
          const routePaths = files.routes.map((r) => unhashUrl(r).replace('route.js', ''));
          const found = routePaths.find((r) => file.includes(r));
          if (found) {
            watcher.emit('route', hashUrl(`${found}route.js`));
          }
        }
      } else if (f.endsWith('.css') && f.includes('svelte-')) {
        files.publicCssFile = makeCssRelative({
          file,
          distElder: settings.distDir,
        });
        watcher.emit('publicCssFile', files.publicCssFile);
      } else if (f === 'hooks.js') {
        watcher.emit('hooks', hashUrl(file));
      } else if (f === 'shortcodes.js') {
        watcher.emit('shortcodes', hashUrl(file));
      } else if (file.includes(settings.ssrComponents)) {
        const idx = files.server.findIndex((f) => f.includes(fixedFile));
        files.server[idx] = hashUrl(fixedFile);
        // console.log('changed server', hashUrl(fixedFile));
        watcher.emit('ssr', hashUrl(fixedFile));
      } else if (file.includes(settings.clientComponents)) {
        const withoutHash = fixedFile.split('.')[0];
        const idx = files.client.findIndex((f) => f.includes(withoutHash));
        const old = files.client[idx];
        files.client[idx] = fixedFile;
        console.log('changed client', fixedFile, old);
        watcher.emit('client', fixedFile);
      } else if (settings.configFiles.some((f) => file === f)) {
        watcher.emit('elder.config', hashUrl(file));
        // } else if (file.endsWith('.css')) {
        // watcher.emit('otherCssFile', makeCssRelative({ file, distElder: settings.distDir }));
      } else if (file.endsWith(`${settings.srcDir}/helpers/index.js`)) {
        watcher.emit('helpers', hashUrl(file));
      } else {
        // console.log('other file', file);
      }

      if (file.includes(settings.srcDir)) {
        watcher.emit('srcChange', file);
      }

      files.all = [...chokFiles.keys()];
    }

    let initialScanComplete = false;
    chok.on('add', (file, stat) => {
      chokFiles.set(file, stat);
      if (initialScanComplete) {
        handleChange(file);
      }
    });

    chok.on('change', (file, stat) => {
      if (!chokFiles.has(file) || chokFiles.get(file).size !== stat.size) {
        chokFiles.set(file, stat);
        handleChange(file);
      }
    });

    chok.on('unlink', (file) => {
      chokFiles.delete(file);
      if (file === files.publicCssFile) files.publicCssFile = '';
    });

    chok.on('ready', () => {
      initialScanComplete = true;
    });
  }

  return { files, watcher };
}
