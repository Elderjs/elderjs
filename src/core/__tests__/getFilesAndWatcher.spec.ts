import path from 'path';
import { expect, test, vi } from 'vitest';
import { InternalFiles } from '../../utils/types';
import { updateFilesFn } from '../getFilesAndWatcher';

const first = [
  'test/elder.config.ts',
  'test/build/build.js',
  'test/build/cleanPublic.js',
  'test/build/hooks.js',
  'test/build/server.js',
  'test/build/shortcodes.js',
  'test/build/tsconfig.tsbuildinfo',
  'test/build/components/BlogTeaser.svelte',
  'test/build/components/Clock.svelte',
  'test/build/components/DisplayHooks.svelte',
  'test/build/components/HookDetail.svelte',
  'test/build/layouts/Layout.svelte',
  'test/build/routes/advanced/1.basic-route.md',
  'test/build/routes/advanced/Advanced.svelte',
  'test/build/routes/advanced/basic-route.md',
  'test/build/routes/advanced/data-plumbing.md',
  'test/build/routes/advanced/hooks.md',
  'test/build/routes/advanced/partial-hydration.md',
  'test/build/routes/advanced/plugins.md',
  'test/build/routes/advanced/prop-compression.md',
  'test/build/routes/advanced/route.js',
  'test/build/routes/advanced/routing.md',
  'test/build/routes/advanced/shortcodes.md',
  'test/build/routes/advanced/ssr.md',
  'test/build/routes/advanced/stacks.md',
  'test/build/routes/blog/Blog.svelte',
  'test/build/routes/blog/getting-started.md',
  'test/build/routes/blog/home.md',
  'test/build/routes/blog/how-to-build-a-blog-with-elderjs.md',
  'test/build/routes/blog/is-elderjs-right-for-you.md',
  'test/build/routes/blog/route.js',
  'test/build/routes/hooks/Hooks.svelte',
  'test/build/routes/hooks/route.js',
  'test/build/routes/simple/Simple.svelte',
  'test/build/routes/simple/route.js',
  'test/build/routes/ssr/Ssr.svelte',
  'test/build/routes/ssr/route.js',
  'test/___ELDER___/compiled/components/BlogTeaser.js',
  'test/___ELDER___/compiled/components/Clock.js',
  'test/___ELDER___/compiled/components/DisplayHooks.js',
  'test/___ELDER___/compiled/components/HookDetail.js',
  'test/___ELDER___/compiled/layouts/Layout.js',
  'test/___ELDER___/compiled/routes/advanced/Advanced.js',
  'test/___ELDER___/compiled/routes/blog/Blog.js',
  'test/___ELDER___/compiled/routes/hooks/Hooks.js',
  'test/___ELDER___/compiled/routes/simple/Simple.js',
  'test/___ELDER___/compiled/routes/ssr/Ssr.js',
  'test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js',
  'test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js',
  'test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js',
  'test/public/_elderjs/svelte/components/Clock.RXDT3EO6.js',
  'test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js',
  'test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js',
  'test/public/style copy.css',
  'test/public/style.css',
  'test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css',
];
const second = [
  'test/elder.config.ts',
  'test/build/build.js',
  'test/build/cleanPublic.js',
  'test/build/hooks.js',
  'test/build/server.js',
  'test/build/shortcodes.js',
  'test/build/tsconfig.tsbuildinfo',
  'test/build/components/BlogTeaser.svelte',
  'test/build/components/Clock.svelte',
  'test/build/components/DisplayHooks.svelte',
  'test/build/components/HookDetail.svelte',
  'test/build/layouts/Layout.svelte',
  'test/build/routes/advanced/1.basic-route.md',
  'test/build/routes/advanced/Advanced.svelte',
  'test/build/routes/advanced/basic-route.md',
  'test/build/routes/advanced/data-plumbing.md',
  'test/build/routes/advanced/hooks.md',
  'test/build/routes/advanced/partial-hydration.md',
  'test/build/routes/advanced/plugins.md',
  'test/build/routes/advanced/prop-compression.md',
  'test/build/routes/advanced/route.js',
  'test/build/routes/advanced/routing.md',
  'test/build/routes/advanced/shortcodes.md',
  'test/build/routes/advanced/ssr.md',
  'test/build/routes/advanced/stacks.md',
  'test/build/routes/blog/Blog.svelte',
  'test/build/routes/blog/getting-started.md',
  'test/build/routes/blog/home.md',
  'test/build/routes/blog/how-to-build-a-blog-with-elderjs.md',
  'test/build/routes/blog/is-elderjs-right-for-you.md',
  'test/build/routes/blog/route.js',
  'test/build/routes/hooks/Hooks.svelte',
  'test/build/routes/hooks/route.js',
  'test/build/routes/simple/Simple.svelte',
  'test/build/routes/simple/route.js',
  'test/build/routes/ssr/Ssr.svelte',
  'test/build/routes/ssr/route.js',
  'test/___ELDER___/compiled/components/BlogTeaser.js',
  'test/___ELDER___/compiled/components/Clock.js',
  'test/___ELDER___/compiled/components/DisplayHooks.js',
  'test/___ELDER___/compiled/components/HookDetail.js',
  'test/___ELDER___/compiled/layouts/Layout.js',
  'test/___ELDER___/compiled/routes/advanced/Advanced.js',
  'test/___ELDER___/compiled/routes/blog/Blog.js',
  'test/___ELDER___/compiled/routes/hooks/Hooks.js',
  'test/___ELDER___/compiled/routes/simple/Simple.js',
  'test/___ELDER___/compiled/routes/ssr/Ssr.js',
  'test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js',
  'test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js',
  'test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js',
  'test/public/_elderjs/svelte/components/Clock.CHANGED.js',
  'test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js',
  'test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js',
  'test/public/style copy.css',
  'test/public/style.css',
  'test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css',
];

vi.mock('fast-glob', () => ({
  default: {
    sync: vi
      .fn()
      .mockImplementationOnce(() => [...first])
      .mockImplementationOnce(() => [...second]),
  },
}));

vi.mock('../../utils/getUniqueId.js', () => ({
  default: () => 'test',
}));

test('#updateFilesFn > Checks pass by reference, finding the right files, single glob', async () => {
  const settings = {
    srcDir: 'test/build',
    distDir: 'test/public',
    ssrComponents: 'test/___ELDER___/compiled/',
    clientComponents: 'test/public/_elderjs/svelte/',
    configFiles: ['test/elder.config.ts'],
  };
  const updateFiles = (files) => () =>
    updateFilesFn({
      files,
      settings,
      paths: [
        `${settings.srcDir}/**/*`,
        `${settings.ssrComponents}/**/*.js`,
        `${settings.clientComponents}/**/*.js`,
        `${settings.distDir}/**/*.css`,
        ...settings.configFiles,
      ],
    });

  const files = {} as InternalFiles;
  const update = updateFiles(files);
  update();
  expect(files).toBe(files);
  expect(files.all).toEqual(first);
  expect(files).toMatchSnapshot();

  update();

  expect(files).toBe(files);
  expect(files.all).toEqual(second);
  expect(files).toMatchSnapshot();
});
