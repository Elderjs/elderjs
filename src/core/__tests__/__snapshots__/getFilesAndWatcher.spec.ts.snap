// Vitest Snapshot v1

exports[`#updateFilesFn > Checks pass by reference, finding the right files, single glob 1`] = `
{
  "all": [
    "test/elder.config.ts",
    "test/build/build.js",
    "test/build/cleanPublic.js",
    "test/build/hooks.js",
    "test/build/server.js",
    "test/build/shortcodes.js",
    "test/build/tsconfig.tsbuildinfo",
    "test/build/components/BlogTeaser.svelte",
    "test/build/components/Clock.svelte",
    "test/build/components/DisplayHooks.svelte",
    "test/build/components/HookDetail.svelte",
    "test/build/layouts/Layout.svelte",
    "test/build/routes/advanced/1.basic-route.md",
    "test/build/routes/advanced/Advanced.svelte",
    "test/build/routes/advanced/basic-route.md",
    "test/build/routes/advanced/data-plumbing.md",
    "test/build/routes/advanced/hooks.md",
    "test/build/routes/advanced/partial-hydration.md",
    "test/build/routes/advanced/plugins.md",
    "test/build/routes/advanced/prop-compression.md",
    "test/build/routes/advanced/route.js",
    "test/build/routes/advanced/routing.md",
    "test/build/routes/advanced/shortcodes.md",
    "test/build/routes/advanced/ssr.md",
    "test/build/routes/advanced/stacks.md",
    "test/build/routes/blog/Blog.svelte",
    "test/build/routes/blog/getting-started.md",
    "test/build/routes/blog/home.md",
    "test/build/routes/blog/how-to-build-a-blog-with-elderjs.md",
    "test/build/routes/blog/is-elderjs-right-for-you.md",
    "test/build/routes/blog/route.js",
    "test/build/routes/hooks/Hooks.svelte",
    "test/build/routes/hooks/route.js",
    "test/build/routes/simple/Simple.svelte",
    "test/build/routes/simple/route.js",
    "test/build/routes/ssr/Ssr.svelte",
    "test/build/routes/ssr/route.js",
    "test/___ELDER___/compiled/components/BlogTeaser.js",
    "test/___ELDER___/compiled/components/Clock.js",
    "test/___ELDER___/compiled/components/DisplayHooks.js",
    "test/___ELDER___/compiled/components/HookDetail.js",
    "test/___ELDER___/compiled/layouts/Layout.js",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js",
    "test/___ELDER___/compiled/routes/blog/Blog.js",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js",
    "test/___ELDER___/compiled/routes/simple/Simple.js",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js",
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.RXDT3EO6.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
    "test/public/style copy.css",
    "test/public/style.css",
    "test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  ],
  "client": [
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.RXDT3EO6.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
  ],
  "hooks": "test/build/hooks.js?hash=test",
  "publicCssFile": "/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  "routes": [
    "test/build/routes/advanced/route.js?hash=test",
    "test/build/routes/blog/route.js?hash=test",
    "test/build/routes/hooks/route.js?hash=test",
    "test/build/routes/simple/route.js?hash=test",
    "test/build/routes/ssr/route.js?hash=test",
  ],
  "server": [
    "test/___ELDER___/compiled/components/BlogTeaser.js?hash=test",
    "test/___ELDER___/compiled/components/Clock.js?hash=test",
    "test/___ELDER___/compiled/components/DisplayHooks.js?hash=test",
    "test/___ELDER___/compiled/components/HookDetail.js?hash=test",
    "test/___ELDER___/compiled/layouts/Layout.js?hash=test",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js?hash=test",
    "test/___ELDER___/compiled/routes/blog/Blog.js?hash=test",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js?hash=test",
    "test/___ELDER___/compiled/routes/simple/Simple.js?hash=test",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js?hash=test",
  ],
  "shortcodes": "test/build/shortcodes.js?hash=test",
}
`;

exports[`#updateFilesFn > Checks pass by reference, finding the right files, single glob 2`] = `
{
  "all": [
    "test/elder.config.ts",
    "test/build/build.js",
    "test/build/cleanPublic.js",
    "test/build/hooks.js",
    "test/build/server.js",
    "test/build/shortcodes.js",
    "test/build/tsconfig.tsbuildinfo",
    "test/build/components/BlogTeaser.svelte",
    "test/build/components/Clock.svelte",
    "test/build/components/DisplayHooks.svelte",
    "test/build/components/HookDetail.svelte",
    "test/build/layouts/Layout.svelte",
    "test/build/routes/advanced/1.basic-route.md",
    "test/build/routes/advanced/Advanced.svelte",
    "test/build/routes/advanced/basic-route.md",
    "test/build/routes/advanced/data-plumbing.md",
    "test/build/routes/advanced/hooks.md",
    "test/build/routes/advanced/partial-hydration.md",
    "test/build/routes/advanced/plugins.md",
    "test/build/routes/advanced/prop-compression.md",
    "test/build/routes/advanced/route.js",
    "test/build/routes/advanced/routing.md",
    "test/build/routes/advanced/shortcodes.md",
    "test/build/routes/advanced/ssr.md",
    "test/build/routes/advanced/stacks.md",
    "test/build/routes/blog/Blog.svelte",
    "test/build/routes/blog/getting-started.md",
    "test/build/routes/blog/home.md",
    "test/build/routes/blog/how-to-build-a-blog-with-elderjs.md",
    "test/build/routes/blog/is-elderjs-right-for-you.md",
    "test/build/routes/blog/route.js",
    "test/build/routes/hooks/Hooks.svelte",
    "test/build/routes/hooks/route.js",
    "test/build/routes/simple/Simple.svelte",
    "test/build/routes/simple/route.js",
    "test/build/routes/ssr/Ssr.svelte",
    "test/build/routes/ssr/route.js",
    "test/___ELDER___/compiled/components/BlogTeaser.js",
    "test/___ELDER___/compiled/components/Clock.js",
    "test/___ELDER___/compiled/components/DisplayHooks.js",
    "test/___ELDER___/compiled/components/HookDetail.js",
    "test/___ELDER___/compiled/layouts/Layout.js",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js",
    "test/___ELDER___/compiled/routes/blog/Blog.js",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js",
    "test/___ELDER___/compiled/routes/simple/Simple.js",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js",
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.CHANGED.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
    "test/public/style copy.css",
    "test/public/style.css",
    "test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  ],
  "client": [
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.CHANGED.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
  ],
  "hooks": "test/build/hooks.js?hash=test",
  "publicCssFile": "/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  "routes": [
    "test/build/routes/advanced/route.js?hash=test",
    "test/build/routes/blog/route.js?hash=test",
    "test/build/routes/hooks/route.js?hash=test",
    "test/build/routes/simple/route.js?hash=test",
    "test/build/routes/ssr/route.js?hash=test",
  ],
  "server": [
    "test/___ELDER___/compiled/components/BlogTeaser.js?hash=test",
    "test/___ELDER___/compiled/components/Clock.js?hash=test",
    "test/___ELDER___/compiled/components/DisplayHooks.js?hash=test",
    "test/___ELDER___/compiled/components/HookDetail.js?hash=test",
    "test/___ELDER___/compiled/layouts/Layout.js?hash=test",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js?hash=test",
    "test/___ELDER___/compiled/routes/blog/Blog.js?hash=test",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js?hash=test",
    "test/___ELDER___/compiled/routes/simple/Simple.js?hash=test",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js?hash=test",
  ],
  "shortcodes": "test/build/shortcodes.js?hash=test",
}
`;

exports[`#updateFilesFn 1`] = `
{
  "all": [
    "test/elder.config.ts",
    "test/build/build.js",
    "test/build/cleanPublic.js",
    "test/build/hooks.js",
    "test/build/server.js",
    "test/build/shortcodes.js",
    "test/build/tsconfig.tsbuildinfo",
    "test/build/components/BlogTeaser.svelte",
    "test/build/components/Clock.svelte",
    "test/build/components/DisplayHooks.svelte",
    "test/build/components/HookDetail.svelte",
    "test/build/layouts/Layout.svelte",
    "test/build/routes/advanced/1.basic-route.md",
    "test/build/routes/advanced/Advanced.svelte",
    "test/build/routes/advanced/basic-route.md",
    "test/build/routes/advanced/data-plumbing.md",
    "test/build/routes/advanced/hooks.md",
    "test/build/routes/advanced/partial-hydration.md",
    "test/build/routes/advanced/plugins.md",
    "test/build/routes/advanced/prop-compression.md",
    "test/build/routes/advanced/route.js",
    "test/build/routes/advanced/routing.md",
    "test/build/routes/advanced/shortcodes.md",
    "test/build/routes/advanced/ssr.md",
    "test/build/routes/advanced/stacks.md",
    "test/build/routes/blog/Blog.svelte",
    "test/build/routes/blog/getting-started.md",
    "test/build/routes/blog/home.md",
    "test/build/routes/blog/how-to-build-a-blog-with-elderjs.md",
    "test/build/routes/blog/is-elderjs-right-for-you.md",
    "test/build/routes/blog/route.js",
    "test/build/routes/hooks/Hooks.svelte",
    "test/build/routes/hooks/route.js",
    "test/build/routes/simple/Simple.svelte",
    "test/build/routes/simple/route.js",
    "test/build/routes/ssr/Ssr.svelte",
    "test/build/routes/ssr/route.js",
    "test/___ELDER___/compiled/components/BlogTeaser.js",
    "test/___ELDER___/compiled/components/Clock.js",
    "test/___ELDER___/compiled/components/DisplayHooks.js",
    "test/___ELDER___/compiled/components/HookDetail.js",
    "test/___ELDER___/compiled/layouts/Layout.js",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js",
    "test/___ELDER___/compiled/routes/blog/Blog.js",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js",
    "test/___ELDER___/compiled/routes/simple/Simple.js",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js",
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.RXDT3EO6.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
    "test/public/style copy.css",
    "test/public/style.css",
    "test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  ],
  "client": [
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.RXDT3EO6.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
  ],
  "hooks": "test/build/hooks.js?hash=test",
  "publicCssFile": "/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  "routes": [
    "test/build/routes/advanced/route.js?hash=test",
    "test/build/routes/blog/route.js?hash=test",
    "test/build/routes/hooks/route.js?hash=test",
    "test/build/routes/simple/route.js?hash=test",
    "test/build/routes/ssr/route.js?hash=test",
  ],
  "server": [
    "test/___ELDER___/compiled/components/BlogTeaser.js?hash=test",
    "test/___ELDER___/compiled/components/Clock.js?hash=test",
    "test/___ELDER___/compiled/components/DisplayHooks.js?hash=test",
    "test/___ELDER___/compiled/components/HookDetail.js?hash=test",
    "test/___ELDER___/compiled/layouts/Layout.js?hash=test",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js?hash=test",
    "test/___ELDER___/compiled/routes/blog/Blog.js?hash=test",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js?hash=test",
    "test/___ELDER___/compiled/routes/simple/Simple.js?hash=test",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js?hash=test",
  ],
  "shortcodes": "test/build/shortcodes.js?hash=test",
}
`;

exports[`#updateFilesFn 2`] = `
{
  "all": [
    "test/elder.config.ts",
    "test/build/build.js",
    "test/build/cleanPublic.js",
    "test/build/hooks.js",
    "test/build/server.js",
    "test/build/shortcodes.js",
    "test/build/tsconfig.tsbuildinfo",
    "test/build/components/BlogTeaser.svelte",
    "test/build/components/Clock.svelte",
    "test/build/components/DisplayHooks.svelte",
    "test/build/components/HookDetail.svelte",
    "test/build/layouts/Layout.svelte",
    "test/build/routes/advanced/1.basic-route.md",
    "test/build/routes/advanced/Advanced.svelte",
    "test/build/routes/advanced/basic-route.md",
    "test/build/routes/advanced/data-plumbing.md",
    "test/build/routes/advanced/hooks.md",
    "test/build/routes/advanced/partial-hydration.md",
    "test/build/routes/advanced/plugins.md",
    "test/build/routes/advanced/prop-compression.md",
    "test/build/routes/advanced/route.js",
    "test/build/routes/advanced/routing.md",
    "test/build/routes/advanced/shortcodes.md",
    "test/build/routes/advanced/ssr.md",
    "test/build/routes/advanced/stacks.md",
    "test/build/routes/blog/Blog.svelte",
    "test/build/routes/blog/getting-started.md",
    "test/build/routes/blog/home.md",
    "test/build/routes/blog/how-to-build-a-blog-with-elderjs.md",
    "test/build/routes/blog/is-elderjs-right-for-you.md",
    "test/build/routes/blog/route.js",
    "test/build/routes/hooks/Hooks.svelte",
    "test/build/routes/hooks/route.js",
    "test/build/routes/simple/Simple.svelte",
    "test/build/routes/simple/route.js",
    "test/build/routes/ssr/Ssr.svelte",
    "test/build/routes/ssr/route.js",
    "test/___ELDER___/compiled/components/BlogTeaser.js",
    "test/___ELDER___/compiled/components/Clock.js",
    "test/___ELDER___/compiled/components/DisplayHooks.js",
    "test/___ELDER___/compiled/components/HookDetail.js",
    "test/___ELDER___/compiled/layouts/Layout.js",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js",
    "test/___ELDER___/compiled/routes/blog/Blog.js",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js",
    "test/___ELDER___/compiled/routes/simple/Simple.js",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js",
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.CHANGED.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
    "test/public/style copy.css",
    "test/public/style.css",
    "test/public/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  ],
  "client": [
    "test/public/_elderjs/svelte/chunks/chunk.BFER7IGE.js",
    "test/public/_elderjs/svelte/chunks/chunk.VLX2NJWY.js",
    "test/public/_elderjs/svelte/components/BlogTeaser.3AUXOWM7.js",
    "test/public/_elderjs/svelte/components/Clock.CHANGED.js",
    "test/public/_elderjs/svelte/components/DisplayHooks.IHTGJQ3I.js",
    "test/public/_elderjs/svelte/components/HookDetail.MCOLEGK2.js",
  ],
  "hooks": "test/build/hooks.js?hash=test",
  "publicCssFile": "/_elderjs/assets/svelte-3bcbf5032efa392596bf0a3f1bdcce7e.css",
  "routes": [
    "test/build/routes/advanced/route.js?hash=test",
    "test/build/routes/blog/route.js?hash=test",
    "test/build/routes/hooks/route.js?hash=test",
    "test/build/routes/simple/route.js?hash=test",
    "test/build/routes/ssr/route.js?hash=test",
  ],
  "server": [
    "test/___ELDER___/compiled/components/BlogTeaser.js?hash=test",
    "test/___ELDER___/compiled/components/Clock.js?hash=test",
    "test/___ELDER___/compiled/components/DisplayHooks.js?hash=test",
    "test/___ELDER___/compiled/components/HookDetail.js?hash=test",
    "test/___ELDER___/compiled/layouts/Layout.js?hash=test",
    "test/___ELDER___/compiled/routes/advanced/Advanced.js?hash=test",
    "test/___ELDER___/compiled/routes/blog/Blog.js?hash=test",
    "test/___ELDER___/compiled/routes/hooks/Hooks.js?hash=test",
    "test/___ELDER___/compiled/routes/simple/Simple.js?hash=test",
    "test/___ELDER___/compiled/routes/ssr/Ssr.js?hash=test",
  ],
  "shortcodes": "test/build/shortcodes.js?hash=test",
}
`;
